### PR TITLE
feat: `w3 delegation create --base64` & `w3 space add <base64>`

### PR DESCRIPTION
**Pull request by olizilla**
**export** a delegation as base64 encoded identity CID with `w3 delegation create --base64`

```shell
$ w3 delegation create did:key:z6MkviAsUfBwegmB57byQ7SZTFtX4jNjo315EvgurjWYoTRX --can 'store/add' --can 'upload/add' --base64
mAYIEAO0OEaJlcm9vdHOAZ3ZlcnNpb24BvQUBcRIgL7w/mAWPOV8Wt/B0ygdaOI+20/ZG8pX+5YzZ5X9U4y6oYXNYRO2hA0CbPDJlxyrorHHdNAUnRUDA4xU7KHgHHstkM8tBxq+6KaQP5xLCknOh9TjkR0S0yuK/fiFxKwRDUHfECFEWQn4DYXZlMC45LjFjYXR0hqJjY2FuZ3NwYWNlLypkd2l0aHg4ZGlkOmtleTp6Nk1raGZ6VHdaSjI4YVJvYkNwNzZ1WFJxenNqSDZHTnUxOFdGd011bWtBRjVvaleiY2NhbmdzdG9yZS8qZHdpdGh4OGRpZDprZXk6ejZNa2hmelR3WkoyOGFSb2JDcDc2dVhScXpzakg2R051MThXRndNdW1rQUY1b2pXomNjYW5odXBsb2FkLypkd2l0aHg4ZGlkOmtleTp6Nk1raGZ6VHdaSjI4YVJvYkNwNzZ1WFJxenNqSDZHTnUxOFdGd011bWtBRjVvaleiY2NhbmhhY2Nlc3MvKmR3aXRoeDhkaWQ6a2V5Ono2TWtoZnpUd1pKMjhhUm9iQ3A3NnVYUnF6c2pINkdOdTE4V0Z3TXVta0FGNW9qV6JjY2FuamZpbGVjb2luLypkd2l0aHg4ZGlkOmtleTp6Nk1raGZ6VHdaSjI4YVJvYkNwNzZ1WFJxenNqSDZHTnUxOFdGd011bWtBRjVvaleiY2Nhbmd1c2FnZS8qZHdpdGh4OGRpZDprZXk6ejZNa2hmelR3WkoyOGFSb2JDcDc2dVhScXpzakg2R051MThXRndNdW1rQUY1b2pXY2F1ZFgi7QHob+19JDMUBs+u1e646vN2MLovQUXA7xJeFs2THUcb+mNleHAaZzctGmNmY3SBoWVzcGFjZaFkbmFtZWV0b290c2Npc3NYIu0BL9X+p4Uyz05zSH0ol8TYPXpwU9EljNRo1O18uYbWlvljcHJmgL0FAXESIC+8P5gFjzlfFrfwdMoHWjiPttP2RvKV/uWM2eV/VOMuqGFzWETtoQNAmzwyZccq6Kxx3TQFJ0VAwOMVOyh4Bx7LZDPLQcavuimkD+cSwpJzofU45EdEtMriv34hcSsEQ1B3xAhRFkJ+A2F2ZTAuOS4xY2F0dIaiY2NhbmdzcGFjZS8qZHdpdGh4OGRpZDprZXk6ejZNa2hmelR3WkoyOGFSb2JDcDc2dVhScXpzakg2R051MThXRndNdW1rQUY1b2pXomNjYW5nc3RvcmUvKmR3aXRoeDhkaWQ6a2V5Ono2TWtoZnpUd1pKMjhhUm9iQ3A3NnVYUnF6c2pINkdOdTE4V0Z3TXVta0FGNW9qV6JjY2FuaHVwbG9hZC8qZHdpdGh4OGRpZDprZXk6ejZNa2hmelR3WkoyOGFSb2JDcDc2dVhScXpzakg2R051MThXRndNdW1rQUY1b2pXomNjYW5oYWNjZXNzLypkd2l0aHg4ZGlkOmtleTp6Nk1raGZ6VHdaSjI4YVJvYkNwNzZ1WFJxenNqSDZHTnUxOFdGd011bWtBRjVvaleiY2NhbmpmaWxlY29pbi8qZHdpdGh4OGRpZDprZXk6ejZNa2hmelR3WkoyOGFSb2JDcDc2dVhScXpzakg2R051MThXRndNdW1rQUY1b2pXomNjYW5ndXNhZ2UvKmR3aXRoeDhkaWQ6a2V5Ono2TWtoZnpUd1pKMjhhUm9iQ3A3NnVYUnF6c2pINkdOdTE4V0Z3TXVta0FGNW9qV2NhdWRYIu0B6G/tfSQzFAbPrtXuuOrzdjC6L0FFwO8SXhbNkx1HG/pjZXhwGmc3LRpjZmN0gaFlc3BhY2WhZG5hbWVldG9vdHNjaXNzWCLtAS/V/qeFMs9Oc0h9KJfE2D16cFPRJYzUaNTtfLmG1pb5Y3ByZoDbAwFxEiB9iHpD1ttdKEQCvBZ8jJBD7Wqw1abOtYwNCKAKYALXMqhhc1hE7aEDQJ7U8I+a4Au/eb10r9T89weG/Nl2jccEUXHs8wq+i2tU0Iaik8KaKvovDqqE57JU8ZoY0JAzOBW7cMLGcV6/UwthdmUwLjkuMWNhdHSComNjYW5pc3RvcmUvYWRkZHdpdGh4OGRpZDprZXk6ejZNa2hmelR3WkoyOGFSb2JDcDc2dVhScXpzakg2R051MThXRndNdW1rQUY1b2pXomNjYW5qdXBsb2FkL2FkZGR3aXRoeDhkaWQ6a2V5Ono2TWtoZnpUd1pKMjhhUm9iQ3A3NnVYUnF6c2pINkdOdTE4V0Z3TXVta0FGNW9qV2NhdWRYIu0B8YzaLs8NDe7oZt6rlpsW6iMh8XsoXZLvkHPqtftYRXZjZXhw9mNmY3SBoWVzcGFjZaFkbmFtZWV0b290c2Npc3NYIu0B6G/tfSQzFAbPrtXuuOrzdjC6L0FFwO8SXhbNkx1HG/pjcHJmgtgqWCUAAXESIC+8P5gFjzlfFrfwdMoHWjiPttP2RvKV/uWM2eV/VOMu2CpYJQABcRIgL7w/mAWPOV8Wt/B0ygdaOI+20/ZG8pX+5YzZ5X9U4y4
```

...yes, what if we put the CAR _in_ the CID! That way we can detect when the input has been truncted, and the screed is self describing... you can paste it into [cid.ipfs.tech](https://cid.ipfs.tech/#mAYIEAO0OEaJlcm9vdHOAZ3ZlcnNpb24BvQUBcRIgL7w/mAWPOV8Wt/B0ygdaOI+20/ZG8pX+5YzZ5X9U4y6oYXNYRO2hA0CbPDJlxyrorHHdNAUnRUDA4xU7KHgHHstkM8tBxq+6KaQP5xLCknOh9TjkR0S0yuK/fiFxKwRDUHfECFEWQn4DYXZlMC45LjFjYXR0hqJjY2FuZ3NwYWNlLypkd2l0aHg4ZGlkOmtleTp6Nk1raGZ6VHdaSjI4YVJvYkNwNzZ1WFJxenNqSDZHTnUxOFdGd011bWtBRjVvaleiY2NhbmdzdG9yZS8qZHdpdGh4OGRpZDprZXk6ejZNa2hmelR3WkoyOGFSb2JDcDc2dVhScXpzakg2R051MThXRndNdW1rQUY1b2pXomNjYW5odXBsb2FkLypkd2l0aHg4ZGlkOmtleTp6Nk1raGZ6VHdaSjI4YVJvYkNwNzZ1WFJxenNqSDZHTnUxOFdGd011bWtBRjVvaleiY2NhbmhhY2Nlc3MvKmR3aXRoeDhkaWQ6a2V5Ono2TWtoZnpUd1pKMjhhUm9iQ3A3NnVYUnF6c2pINkdOdTE4V0Z3TXVta0FGNW9qV6JjY2FuamZpbGVjb2luLypkd2l0aHg4ZGlkOmtleTp6Nk1raGZ6VHdaSjI4YVJvYkNwNzZ1WFJxenNqSDZHTnUxOFdGd011bWtBRjVvaleiY2Nhbmd1c2FnZS8qZHdpdGh4OGRpZDprZXk6ejZNa2hmelR3WkoyOGFSb2JDcDc2dVhScXpzakg2R051MThXRndNdW1rQUY1b2pXY2F1ZFgi7QHob+19JDMUBs+u1e646vN2MLovQUXA7xJeFs2THUcb+mNleHAaZzctGmNmY3SBoWVzcGFjZaFkbmFtZWV0b290c2Npc3NYIu0BL9X+p4Uyz05zSH0ol8TYPXpwU9EljNRo1O18uYbWlvljcHJmgL0FAXESIC+8P5gFjzlfFrfwdMoHWjiPttP2RvKV/uWM2eV/VOMuqGFzWETtoQNAmzwyZccq6Kxx3TQFJ0VAwOMVOyh4Bx7LZDPLQcavuimkD+cSwpJzofU45EdEtMriv34hcSsEQ1B3xAhRFkJ+A2F2ZTAuOS4xY2F0dIaiY2NhbmdzcGFjZS8qZHdpdGh4OGRpZDprZXk6ejZNa2hmelR3WkoyOGFSb2JDcDc2dVhScXpzakg2R051MThXRndNdW1rQUY1b2pXomNjYW5nc3RvcmUvKmR3aXRoeDhkaWQ6a2V5Ono2TWtoZnpUd1pKMjhhUm9iQ3A3NnVYUnF6c2pINkdOdTE4V0Z3TXVta0FGNW9qV6JjY2FuaHVwbG9hZC8qZHdpdGh4OGRpZDprZXk6ejZNa2hmelR3WkoyOGFSb2JDcDc2dVhScXpzakg2R051MThXRndNdW1rQUY1b2pXomNjYW5oYWNjZXNzLypkd2l0aHg4ZGlkOmtleTp6Nk1raGZ6VHdaSjI4YVJvYkNwNzZ1WFJxenNqSDZHTnUxOFdGd011bWtBRjVvaleiY2NhbmpmaWxlY29pbi8qZHdpdGh4OGRpZDprZXk6ejZNa2hmelR3WkoyOGFSb2JDcDc2dVhScXpzakg2R051MThXRndNdW1rQUY1b2pXomNjYW5ndXNhZ2UvKmR3aXRoeDhkaWQ6a2V5Ono2TWtoZnpUd1pKMjhhUm9iQ3A3NnVYUnF6c2pINkdOdTE4V0Z3TXVta0FGNW9qV2NhdWRYIu0B6G/tfSQzFAbPrtXuuOrzdjC6L0FFwO8SXhbNkx1HG/pjZXhwGmc3LRpjZmN0gaFlc3BhY2WhZG5hbWVldG9vdHNjaXNzWCLtAS/V/qeFMs9Oc0h9KJfE2D16cFPRJYzUaNTtfLmG1pb5Y3ByZoDbAwFxEiB9iHpD1ttdKEQCvBZ8jJBD7Wqw1abOtYwNCKAKYALXMqhhc1hE7aEDQJ7U8I+a4Au/eb10r9T89weG/Nl2jccEUXHs8wq+i2tU0Iaik8KaKvovDqqE57JU8ZoY0JAzOBW7cMLGcV6/UwthdmUwLjkuMWNhdHSComNjYW5pc3RvcmUvYWRkZHdpdGh4OGRpZDprZXk6ejZNa2hmelR3WkoyOGFSb2JDcDc2dVhScXpzakg2R051MThXRndNdW1rQUY1b2pXomNjYW5qdXBsb2FkL2FkZGR3aXRoeDhkaWQ6a2V5Ono2TWtoZnpUd1pKMjhhUm9iQ3A3NnVYUnF6c2pINkdOdTE4V0Z3TXVta0FGNW9qV2NhdWRYIu0B8YzaLs8NDe7oZt6rlpsW6iMh8XsoXZLvkHPqtftYRXZjZXhw9mNmY3SBoWVzcGFjZaFkbmFtZWV0b290c2Npc3NYIu0B6G/tfSQzFAbPrtXuuOrzdjC6L0FFwO8SXhbNkx1HG/pjcHJmgtgqWCUAAXESIC+8P5gFjzlfFrfwdMoHWjiPttP2RvKV/uWM2eV/VOMu2CpYJQABcRIgL7w/mAWPOV8Wt/B0ygdaOI+20/ZG8pX+5YzZ5X9U4y4) and it'll tell you it's CAR flavour identity hashed bytes!

**import** a space from a stringified proof

```shell
$ w3 space add $PROOF
did:key:z6MkhfzTwZJ28aRobCp76uXRqzsjH6GNu18WFwMumkAF5ojW
```

see: https://github.com/web3-storage/w3cli/issues/154#issuecomment-1885042715

## Usage

**on your machine**

- with `@web3-storage/w3cli` installed.
- Set the space you want CI to upload to as the current space

```shell
# create a key for ci
$ w3 key create --json > ci.json

# create a proof for that key. copy paste it to env var in CI
$ w3 delegation create $(jq -r .did ci.json) -c 'store/add' -c 'upload/add' --base64
mA...
```

**in CI**

- install `@web3-storage/w3cli`. No need to login, we pass it the proof to use.
- copy paste the key we generated above `(jq -r .key ci.json)` as a secret in your ci workflow. _KEEP IT SECRET!_
- copy paste the proof we generated above as a variable or secret in your ci workflow. _it's ok to share this, only the holder of the secret key can use it._

```shell
# set key (jq -r .key ci.json) in env so w3cli uses it instead of generating one
$ W3_PRINCIPAL=${{ secrets.w3key }}

# import the space from the stringified proof
$ w3 space add ${{ vars.w3proof }}

# upload yer stuff
$ w3 upload ./my/cool/stuff
```

License: MIT

### CODE REVIEW

**Comment by olizilla** [comment]
`baseemoji256` was also considered, but wow

```shell
$ w3 delegation create did:key:z6MkviAsUfBwegmB57byQ7SZTFtX4jNjo315EvgurjWYoTRX --can 'store/add' --can 'upload/add' --stringify
🚀🪐🎂🌌🚀🥀🌏☀🧡✋🥺😓😓😈🌈😕😝💦✋🥺🌈🌟😓😻🪐🌱🌑🪐✅💻😅👌💟🙌🤫🌑😛✌🥳😂✈💡😈😷🌓😞💞😛🤬🤧🤮💓👐🤙📣🙇👑🍒🙇☝💋👆☺✊🤤🌈💥💛🥀🌿🛰😋🙆😱😔✋😨😆🌴😶✅🤯😇🌑💙🙂😋🥴👆💿🤷😢😣🌓😘🕺😳😎🕺💗⬇💰🤓🤔⚡🌎❄💻➡🌼🌈🌿🤐💞🎤🤩💛🚨😷💆👶😠👏✅🙄🌌😏😒✔💎🌔🤭😂💚😠🛰🤤💦✋🤗☺✌☺💜💃🤤😈😈💝🧡💃💃🤤😻😝🌈⭐🤤💃✋👌😆😳✔🌟😈😴😣💞😳🌟😳✨🙃✋🏃✨💐🤦🙈🙃😴😚💐💋✔😞🖤😔💞🤤❣😓👉😏⭐🎉🤦🤘💥❣✅💐🌈😬😄🤦🤩👇🤘💜💞😑💓✔🙈🤘🌷🙃💗💓🌹😓😬😑🧡💃💃🤤😻😝🌈😈😓🥺✋👌😆😳✔🌟😈😴😣💞😳🌟😳✨🙃✋🏃✨💐🤦🙈🙃😴😚💐💋✔😞🖤😔💞🤤❣😓👉😏⭐🎉🤦🤘💥❣✅💐🌈😬😄🤦🤩👇🤘💜💞😑💓✔🙈🤘🌷🙃💗💓🌹😓😬😑🧡💃💃🤤😻😴🤘⭐🍀😓🤤😳👌😆😳✔🌟😈😴😣💞😳🌟😳✨🙃✋🏃✨💐🤦🙈🙃😴😚💐💋✔😞🖤😔💞🤤❣😓👉😏⭐🎉🤦🤘💥❣✅💐🌈😬😄🤦🤩👇🤘💜💞😑💓✔🙈🤘🌷🙃💗💓🌹😓😬😑🧡💃💃🤤😻😴🤤💃💃✋🌈🌈👌😆😳✔🌟😈😴😣💞😳🌟😳✨🙃✋🏃✨💐🤦🙈🙃😴😚💐💋✔😞🖤😔💞🤤❣😓👉😏⭐🎉🤦🤘💥❣✅💐🌈😬😄🤦🤩👇🤘💜💞😑💓✔🙈🤘🌷🙃💗💓🌹😓😬😑🧡💃💃🤤😻😬😚🌟🍀✋💃😓🌟😻👌😆😳✔🌟😈😴😣💞😳🌟😳✨🙃✋🏃✨💐🤦🙈🙃😴😚💐💋✔😞🖤😔💞🤤❣😓👉😏⭐🎉🤦🤘💥❣✅💐🌈😬😄🤦🤩👇🤘💜💞😑💓✔🙈🤘🌷🙃💗💓🌹😓😬😑🧡💃💃🤤😻😝🤘🌈🤤😝✋👌😆😳✔🌟😈😴😣💞😳🌟😳✨🙃✋🏃✨💐🤦🙈🙃😴😚💐💋✔😞🖤😔💞🤤❣😓👉😏⭐🎉🤦🤘💥❣✅💐🌈😬😄🤦🤩👇🤘💜💞😑💓✔🙈🤘🌷🙃💗💓🌹😓😬😑💃🤤🤘😳💥😁🥀🪐🌴😓🥀💘🥰😎💾🌒😵🚶🔵🤢🎀🐸📸💦🤗🤓👌💗🙂🥴👅💻👊😂🙅😫😭🤩🙏🍎💃✋😣⭐😊😝🎉😉😊💃😚💃😈🌺🌿✋🌈⭐🤤💃✋🌿😳😻🤤🌷✋✋😈😓😓😈🌈💃🌟🌈🌈💥😁🥀🪐👌🔵📣❌🖕😔😵👇🌈😄💘😢🏆💎🐾😌💐⭐😜🤲💔👑📌😴📌🥀🎊🍺💝💅☕🚩💃⭐🥺😚😕🌱🌑🪐✅💻😅👌💟🙌🤫🌑😛✌🥳😂✈💡😈😷🌓😞💞😛🤬🤧🤮💓👐🤙📣🙇👑🍒🙇☝💋👆☺✊🤤🌈💥💛🥀🌿🛰😋🙆😱😔✋😨😆🌴😶✅🤯😇🌑💙🙂😋🥴👆💿🤷😢😣🌓😘🕺😳😎🕺💗⬇💰🤓🤔⚡🌎❄💻➡🌼🌈🌿🤐💞🎤🤩💛🚨😷💆👶😠👏✅🙄🌌😏😒✔💎🌔🤭😂💚😠🛰🤤💦✋🤗☺✌☺💜💃🤤😈😈💝🧡💃💃🤤😻😝🌈⭐🤤💃✋👌😆😳✔🌟😈😴😣💞😳🌟😳✨🙃✋🏃✨💐🤦🙈🙃😴😚💐💋✔😞🖤😔💞🤤❣😓👉😏⭐🎉🤦🤘💥❣✅💐🌈😬😄🤦🤩👇🤘💜💞😑💓✔🙈🤘🌷🙃💗💓🌹😓😬😑🧡💃💃🤤😻😝🌈😈😓🥺✋👌😆😳✔🌟😈😴😣💞😳🌟😳✨🙃✋🏃✨💐🤦🙈🙃😴😚💐💋✔😞🖤😔💞🤤❣😓👉😏⭐🎉🤦🤘💥❣✅💐🌈😬😄🤦🤩👇🤘💜💞😑💓✔🙈🤘🌷🙃💗💓🌹😓😬😑🧡💃💃🤤😻😴🤘⭐🍀😓🤤😳👌😆😳✔🌟😈😴😣💞😳🌟😳✨🙃✋🏃✨💐🤦🙈🙃😴😚💐💋✔😞🖤😔💞🤤❣😓👉😏⭐🎉🤦🤘💥❣✅💐🌈😬😄🤦🤩👇🤘💜💞😑💓✔🙈🤘🌷🙃💗💓🌹😓😬😑🧡💃💃🤤😻😴🤤💃💃✋🌈🌈👌😆😳✔🌟😈😴😣💞😳🌟😳✨🙃✋🏃✨💐🤦🙈🙃😴😚💐💋✔😞🖤😔💞🤤❣😓👉😏⭐🎉🤦🤘💥❣✅💐🌈😬😄🤦🤩👇🤘💜💞😑💓✔🙈🤘🌷🙃💗💓🌹😓😬😑🧡💃💃🤤😻😬😚🌟🍀✋💃😓🌟😻👌😆😳✔🌟😈😴😣💞😳🌟😳✨🙃✋🏃✨💐🤦🙈🙃😴😚💐💋✔😞🖤😔💞🤤❣😓👉😏⭐🎉🤦🤘💥❣✅💐🌈😬😄🤦🤩👇🤘💜💞😑💓✔🙈🤘🌷🙃💗💓🌹😓😬😑🧡💃💃🤤😻😝🤘🌈🤤😝✋👌😆😳✔🌟😈😴😣💞😳🌟😳✨🙃✋🏃✨💐🤦🙈🙃😴😚💐💋✔😞🖤😔💞🤤❣😓👉😏⭐🎉🤦🤘💥❣✅💐🌈😬😄🤦🤩👇🤘💜💞😑💓✔🙈🤘🌷🙃💗💓🌹😓😬😑💃🤤🤘😳💥😁🥀🪐🌴😓🥀💘🥰😎💾🌒😵🚶🔵🤢🎀🐸📸💦🤗🤓👌💗🙂🥴👅💻👊😂🙅😫😭🤩🙏🍎💃✋😣⭐😊😝🎉😉😊💃😚💃😈🌺🌿✋🌈⭐🤤💃✋🌿😳😻🤤🌷✋✋😈😓😓😈🌈💃🌟🌈🌈💥😁🥀🪐👌🔵📣❌🖕😔😵👇🌈😄💘😢🏆💎🐾😌💐⭐😜🤲💔👑📌😴📌🥀🎊🍺💝💅☕🚩💃⭐🥺😚😕🤑🛰🪐✅💻😅💘😹💐😏💅🤑🤪😢💛☄💟😂🎊👑🔴😏🥀😬🍓🔵🎈😟💨👑🌍🌔😲🌖😥☄🧐😔✊🤤🌈💥💛🥀🌿🛰😋🐶📌💡😛😮💧🌗👶🏃🌱😈💰📌👼🎼🌓💝👼🍒💦🎵😨🌌🤭✅📍📸🌖😖💀🙃💋👎💝🧡😫➡😮😆🍎👌🌏😰😐❄🤟💋💩😮😍👎🔴😎💞💿😙⭐➡⬇✅👊👶😜🌗🤤💦✋🤗☺✌☺💜💃🤤😈😈🎂🧡💃💃🤤😻🌟🌈😈😓🥺✋👌🤤😳😳😳✔🌟😈😴😣💞😳🌟😳✨🙃✋🏃✨💐🤦🙈🙃😴😚💐💋✔😞🖤😔💞🤤❣😓👉😏⭐🎉🤦🤘💥❣✅💐🌈😬😄🤦🤩👇🤘💜💞😑💓✔🙈🤘🌷🙃💗💓🌹😓😬😑🧡💃💃🤤😻😬🤘⭐🍀😓🤤😳👌🤤😳😳😳✔🌟😈😴😣💞😳🌟😳✨🙃✋🏃✨💐🤦🙈🙃😴😚💐💋✔😞🖤😔💞🤤❣😓👉😏⭐🎉🤦🤘💥❣✅💐🌈😬😄🤦🤩👇🤘💜💞😑💓✔🙈🤘🌷🙃💗💓🌹😓😬😑💃🤤🤘😳💥😁🥀🪐💩👑😗☺😵🌍🌍🤢🌴😚🐷🤨☕🙆😂🐸🔥👏💩☹😢🤪🌼👅🔴🌈🐸💨🍊💥🙂💦💃✋😣⭐🤮💃😚💃😈🌺🌿✋🌈⭐🤤💃✋🌿😳😻🤤🌷✋✋😈😓😓😈🌈💃🌟🌈🌈💥😁🥀🪐🌴😓🥀💘🥰😎💾🌒😵🚶🔵🤢🎀🐸📸💦🤗🤓👌💗🙂🥴👅💻👊😂🙅😫😭🤩🙏🍎💃⭐🥺😚🎂🐾😆💥💔🚀🪐✅💻😅👌💟🙌🤫🌑😛✌🥳😂✈💡😈😷🌓😞💞😛🤬🤧🤮💓👐🤙📣🙇👑🍒🙇☝💋👆☺🐾😆💥💔🚀🪐✅💻😅👌💟🙌🤫🌑😛✌🥳😂✈💡😈😷🌓😞💞😛🤬🤧🤮💓👐🤙📣🙇👑🍒🙇☝💋👆☺⏎  
```

**Comment by olizilla** [comment]
Just base64 mutlibase encoding was considered, but the identity cid wrapping lets use specify that it's a CAR via the cid codec for self describing goodness...

...and as it gets length prefixed we can detect truncation, which for long text screeds seems like a win.

File: index.js [hunk 1]
```diff
@@ -253,13 +256,36 @@ export async function createSpace(name) {
...
+      process.exit(1)
+    }
+    const delegation = await readProofFromBytes(cid.multihash.digest)
+    const space = await client.addSpace(delegation)
+    console.log(space.did())
+
+  } else {
+    const delegation = await readProof(proofPathOrCid)
+    const space = await client.addSpace(delegation)
+    console.log(space.did())
```
Comment (alanshaw) [inline index.js:287]: Would be nice to not repeat these 2 lines...

File: lib.js [hunk 2]
```diff
@@ -166,6 +166,29 @@ export async function readProof(path) {
...
+    process.exit(1)
+  }
+  try {
+    // @ts-expect-error
+    return importDAG(blocks)
+  } catch (/** @type {any} */ err) {
+    console.error(`Error: failed to import proof: ${err.message}`)
+    process.exit(1)
+  }
+}
```
Comment (alanshaw) [inline lib.js:190]: `readProof` could maybe call this...

**Review by alanshaw — APPROVED** [review]
This is great! I'm not 100% sold on "stringify" as the option that causes it to encode it to string...but failing to think of something better `--string`? `--output-encoding=base64`? 🤷‍♂️

**Review by vasco-santos — APPROVED** [review]
LGTM ✨

**Review by Fool256 — APPROVED** [review]

**Review by Fool256 — APPROVED** [review]

**Pull request merged**